src/Control/Lens/Cons.hs

```diff
@@ -319,6 +323,42 @@ _tail :: Cons s s a a => Traversal' s s
 _tail = _Cons._2
 {-# INLINE _tail #-}
 
+-- | Convert one type with a 'Cons' instance into the other.
+--
+-- Rewrite rules are provided for efficient conversion between
+-- 'String' and 'Text', @['Word8']@ and 'ByteString', and lazy and
+-- strict 'Text' and 'ByteString'.  Programs must be compiled
+-- with @-O@ to use them.
+--
+-- If @(recons . recons)@ exists it should obey:
+--
+-- > recons . recons ≡ id
+--
+recons :: (Cons s1 s1 a a, Cons s2 s2 a a, AsEmpty s2) => Getter s1 s2
```

Comment:
I originally had that.  @tonymorris urged the change because it would be possible to write 'Cons' instances that result in an optic that does not obey `Iso` laws.  Such an instance would be daft, but still, the types do not guarantee it.

That said, there is precedent in *lens* of unlawful values.  I defer to the *lens* maintainer(s) wishes on this.